AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -97,74 +182,299 @@ def AtomEditorComponents_Light_AddedToEntity():
         light_entity = EditorEntity.create_editor_entity(AtomComponentProperties.light())
         Report.critical_result(Tests.light_creation, light_entity.exists())
 
-        # 2. Add Light component to the Light entity.
-        light_component = light_entity.add_component(AtomComponentProperties.light())
-        Report.critical_result(Tests.light_component, light_entity.has_component(AtomComponentProperties.light()))
+        # Cycle through light types to test component properties.
+        for light_type in LIGHT_TYPES.keys():
 
-        # 3. UNDO the entity creation and component addition.
-        # -> UNDO component addition.
-        general.undo()
-        # -> UNDO naming entity.
-        general.undo()
-        # -> UNDO selecting entity.
-        general.undo()
-        # -> UNDO entity creation.
-        general.undo()
-        general.idle_wait_frames(1)
-        Report.result(Tests.creation_undo, not light_entity.exists())
+            # 2. Remove existing light component from entity.
+            if light_entity.has_component(AtomComponentProperties.light()):
+                light_entity.remove_component(AtomComponentProperties.light())
+                Report.critical_result(Tests.light_component_removal,
+                                       not light_entity.has_component(AtomComponentProperties.light()))
 
-        # 4. REDO the entity creation and component addition.
-        # -> REDO entity creation.
-        general.redo()
-        # -> REDO selecting entity.
-        general.redo()
-        # -> REDO naming entity.
-        general.redo()
-        # -> REDO component addition.
-        general.redo()
-        general.idle_wait_frames(1)
-        Report.result(Tests.creation_redo, light_entity.exists())
+            # 3. Add Light component to the Light entity.
+            light_component = light_entity.add_component(AtomComponentProperties.light())
+            if iterations == 0:
+                Report.critical_result(Tests.light_component,
+                                       light_entity.has_component(AtomComponentProperties.light()))
 
-        # 5. Cycle through all light types.
-        for light_type in LIGHT_TYPES.keys():
+                # 4. UNDO the entity creation and component addition.
```

Comment:
from this line to line 225:
why is there indented further than the calls made above it? can't you just align the spacing with the code above it? that would probably also remove the need for this `iterations` variable which really shouldn't be needed at all (once all `LIGHT_TYPES.keys()` have been iterated upon the loop ends, there shouldn't be need for a tracking variable like `iterations` unless im missing something)

i have a feeling you may have gotten looped because you nested this code inside existing code - try to align it all with the same spacing as the above code, i.e:
```
 # 3. Add Light component to the Light entity.
            light_component = light_entity.add_component(AtomComponentProperties.light())
            Report.critical_result(Tests.light_component,
                                   light_entity.has_component(AtomComponentProperties.light()))

            # 4. UNDO the entity creation and component addition.
            # -> UNDO component addition.
            general.undo()
            # -> UNDO naming entity.
            general.undo()
            # -> UNDO selecting entity.
            general.undo()
# Etc. you get the idea, just align the spacing in the script
```

you can just remove the unknown value from `LIGHT_TYPES`  using LIGHT_TYPES.pop('unknown') before iterating on the `LIGHT_TYPE.keys()` and that would be cleaner than using this `iterations` variable:

Reply:
I removed the entity undo/redo checks entirely, and added an entity.remove_component() and an undo for that action outside of the loop.

It creates a minor redundancy in that once the loop starts it removes/creates a new light component, but it also removes unnecessary undo/redo actions (since we're testing the component, not the entity itself)

Reply:
use light_component.remove() instead of light_entity.remove_component(AtomComponentProperties.light())

Reply:
changed to light_component.remove()